src/app/file-upload-modal/file-upload-modal.component.ts

```diff
@@ -0,0 +1,225 @@
+import { Component, OnInit, ChangeDetectorRef, ElementRef, ViewChild, HostListener } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { NgxSmartModalService } from 'ngx-smart-modal';
+import { Utils } from 'app/shared/utils/utils';
+import { uniqBy } from 'lodash';
+import { UploadedFile } from './types';
+import { ModalData } from 'app/shared/types/modal';
+
+
+@Component({
+  selector: 'app-file-upload-modal',
+  templateUrl: './file-upload-modal.component.html',
+  styleUrls: ['./file-upload-modal.component.scss']
+})
+export class FileUploadModalComponent implements OnInit {
+  closeDialog: boolean;
+  modalData: ModalData;
+  uploadedFiles: UploadedFile[];
+  file: UploadedFile;
+  fileList: UploadedFile[];
+  fileExt: string;
+  selectedFiles: UploadedFile[];
+  @ViewChild("fileBrowser") fileBrowser: ElementRef;
+  @ViewChild("fileInfoForm") fileInfoForm: ElementRef;
+
+  constructor(
+    public utils: Utils,
+    private ngxSmartModalService: NgxSmartModalService,
+    private _changeDetectorRef: ChangeDetectorRef,
+  ) { }
+
+  ngOnInit() {
+    this.closeDialog = false;
+    this.selectedFiles = [];
+    this.fileList = [];
+    this.uploadedFiles = [];
+    this.modalData = this.ngxSmartModalService.getModalData('file-upload-modal');
+
+    if (this.modalData.fileExt) {
+      this.fileExt = this.modalData.fileExt;
+    } else {
+      this.fileExt = 'jpg, jpeg, gif, png, bmp, doc, docx, xls, xlsx, ppt, pptx, pdf, txt, zip';
```

Comment:
does this just show the files, or does this also play a role, in allowing... Lets be sure we want zip, as we know lots of bad guys can hang out in a zip file.
Also just making sure we are doing sanity checks for file uploads to make sure they are the file they say they are, this is where a lot of exploits come from.  Its not the kind of thing we want to be re-inventing ourselfs.

Reply:
.zip are required for ShapeFile upload, if this component is going to be used across the board.

Reply:
this is what i thought... oh well it is what it is, as long as this dialog or upload can't be accessed via non authenticated / authorized user, this shouldn't be too big of an issue.

Reply:
Thanks for the comment, good points.
- zip files are required for shapefiles
- I'm only leveraging the existing app file upload functionality and giving it a nicer interface. To limit the scope of this ticket, I'm not diving into it but relying on it to do its job for now.

In terms of sanity checks, the file uploader doesn't allow extensions that aren't in the list. The user will try to select a file, they will get an error, and the file won't be put into memory. I'm curious to know if there's anything else we should be doing though. Please let me know.